File: Principal/Html/Componente/Activity/ActHome.cs
﻿using NetZ.Web.Html;
using NetZ.Web.Html.Componente.Botao;
using NetZ.Web.Server.Arquivo.Css;

namespace Sentinela.Html.Componente.Activity
{
    internal class ActHome : ActSentinelaBase
    {
        #region Constantes

        #endregion Constantes

        #region Atributos

        private BotaoCircular _btnAdicionar;
        private Div _divLista;

        private BotaoCircular btnAdicionar
        {
            get
            {
                if (_btnAdicionar != null)
                {
                    return _btnAdicionar;
                }

                _btnAdicionar = new BotaoCircular();

                return _btnAdicionar;
            }
        }

        private Div divLista
        {
            get
            {
                if (_divLista != null)
                {
                    return _divLista;
                }

                _divLista = new Div();

                return _divLista;
            }
        }

        #endregion Atributos

        #region Construtores

        #endregion Construtores

        #region Métodos

        protected override void addLayoutFixo(JavaScriptTag tagJs)
        {
            base.addLayoutFixo(tagJs);

            tagJs.addLayoutFixo(typeof(DivServidorTile));
        }

        protected override void inicializar()
        {
            base.inicializar();

            this.btnAdicionar.enmTamanho = BotaoCircular.EnmTamanho.GRANDE;
        }

        protected override void montarLayout()
        {
            base.montarLayout();

            this.btnAdicionar.setPai(this);

            this.divLista.setPai(this);
        }

        protected override void setCss(CssArquivoBase css)
        {
            base.setCss(css);

            this.addCss(css.setDisplay("none"));

            this.btnAdicionar.addCss(css.setBottom(25));
            this.btnAdicionar.addCss(css.setPosition("absolute"));
            this.btnAdicionar.addCss(css.setRight(25));
        }

        protected override void setStrId(string strId)
        {
            base.setStrId(strId);

            this.btnAdicionar.strId = (strId + "_btnAdicionar");
            this.divLista.strId = (strId + "_divLista");
        }

        #endregion Métodos

        #region Eventos

        #endregion Eventos
    }
}

File: Sentinela.TypeScript/html/componente/activity/ActHome.ts
/// <reference path="../../../../Web.TypeScript/erro/Erro.ts"/>
/// <reference path="../../../../Web.TypeScript/html/componente/botao/BotaoCircular.ts"/>
/// <reference path="../../../../Web.TypeScript/html/Div.ts"/>
/// <reference path="../../../../Web.TypeScript/Objeto.ts"/>
/// <reference path="../../../../Web.TypeScript/OnClickListener.ts"/>
/// <reference path="../../../../Web.TypeScript/Utils.ts"/>

module Sentinela
{
    // #region Importações

    import BotaoCircular = Web.BotaoCircular;
    import Div = Web.Div;
    import Erro = Web.Erro;
    import Objeto = Web.Objeto;
    import OnClickListener = Web.OnClickListener;
    import Utils = Web.Utils;

    // #endregion Importações

    // #region Enumerados
    // #endregion Enumerados

    export class ActHome extends ActSentinelaBase implements OnClickListener
    {
        // #region Constantes
        // #endregion Constantes

        // #region Atributos

        private _arrDivServidorGrupo: Array<DivServidorGrupo>;
        private _btnAdicionar: BotaoCircular;
        private _divLista: Div;

        private get arrDivServidorGrupo(): Array<DivServidorGrupo>
        {
            if (this._arrDivServidorGrupo != null)
            {
                return this._arrDivServidorGrupo;
            }

            this._arrDivServidorGrupo = new Array<DivServidorGrupo>();

            return this._arrDivServidorGrupo;
        }

        private get btnAdicionar(): BotaoCircular
        {
            if (this._btnAdicionar != null)
            {
                return this._btnAdicionar;
            }

            this._btnAdicionar = new BotaoCircular(this.strId + "_btnAdicionar");

            return this._btnAdicionar;
        }

        private get divLista(): Div
        {
            if (this._divLista != null)
            {
                return this._divLista;
            }

            this._divLista = new Div(this.strId + "_divLista");

            return this._divLista;
        }

        // #endregion Atributos

        // #region Construtores

        constructor()
        {
            super(ActHome.name);
        }

        // #endregion Construtores

        // #region Métodos

        private adicionar(): void
        {
            PagPrincipal.i.abrirActServidorCadastro();
        }

        private carregarLista(): void
        {
            if (AppSentinela.i.arrObjServidor.length < 1)
            {
                return;
            }

            this.divLista.strConteudo = null;

            AppSentinela.i.arrObjServidor.forEach((o) => this.carregarListaItem(o));
        }

        private carregarListaItem(objServidor: ServidorDominio): void
        {
            if (objServidor == null)
            {
                return;
            }

            if (Utils.getBooStrVazia(objServidor.strNome))
            {
                return;
            }

            var divServidorTile = new DivServidorTile(this, objServidor);

            this.divLista.addHtml(divServidorTile.strLayoutFixo);

            divServidorTile.iniciar();
        }

        protected inicializar(): void
        {
            super.inicializar();

            this.carregarLista();
        }

        protected setEventos(): void
        {
            super.setEventos();

            this.btnAdicionar.addEvtOnClickListener(this);
        }

        // #endregion Métodos

        // #region Eventos

        public onClick(objSender: Objeto, arg: JQueryEventObject): void
        {
            try
            {
                switch (objSender)
                {
                    case this.btnAdicionar:
                        this.adicionar();
                        return;
                }
            }
            catch (ex)
            {
                new Erro("Algo deu errado.", ex);
            }
        }

        // #endregion Eventos
    }
}